modules/gridBidAdapter.js

```diff
@@ -224,75 +129,403 @@ function _getBidFromResponse(respItem) {
+  };
+
+  if (pageKeywords) {
+    payload.keywords = JSON.stringify(pageKeywords);
+  }
+
+  if (bidderRequest) {
+    if (bidderRequest.refererInfo && bidderRequest.refererInfo.referer) {
+      payload.u = bidderRequest.refererInfo.referer;
+    }
+    if (bidderRequest.timeout) {
+      payload.wtimeout = bidderRequest.timeout;
+    }
+    if (bidderRequest.gdprConsent) {
+      if (bidderRequest.gdprConsent.consentString) {
+        payload.gdpr_consent = bidderRequest.gdprConsent.consentString;
+      }
+      payload.gdpr_applies =
+        (typeof bidderRequest.gdprConsent.gdprApplies === 'boolean')
+          ? Number(bidderRequest.gdprConsent.gdprApplies) : 1;
+    }
+    if (bidderRequest.uspConsent) {
+      payload.us_privacy = bidderRequest.uspConsent;
+    }
+  }
+
+  return {
+    method: 'GET',
+    url: ENDPOINT_URL,
+    data: utils.parseQueryStringParameters(payload).replace(/\&$/, ''),
+    bidsMap: bidsMap
+  }
+}
+
+function buildNewRequest(validBidRequests, bidderRequest) {
+  let pageKeywords = null;
+  let jwpseg = null;
+  let content = null;
+  let schain = null;
+  let userId = null;
+  let user = null;
+  let userExt = null;
+  let {bidderRequestId, auctionId, gdprConsent, uspConsent, timeout, refererInfo} = bidderRequest;
+
+  const referer = refererInfo ? encodeURIComponent(refererInfo.referer) : '';
+  const imp = [];
+  const bidsMap = {};
+
+  validBidRequests.forEach((bid) => {
+    if (!bidderRequestId) {
+      bidderRequestId = bid.bidderRequestId;
+    }
+    if (!auctionId) {
+      auctionId = bid.auctionId;
+    }
+    if (!schain) {
+      schain = bid.schain;
+    }
+    if (!userId) {
+      userId = bid.userId;
+    }
+    const {params: {uid, keywords}, mediaTypes, bidId, adUnitCode, realTimeData} = bid;
+    bidsMap[bidId] = bid;
+    if (!pageKeywords && !utils.isEmpty(keywords)) {
+      pageKeywords = utils.transformBidderParamKeywords(keywords);
+    }
+    if (realTimeData && realTimeData.jwTargeting) {
+      if (!jwpseg && realTimeData.jwTargeting.segments) {
+        jwpseg = realTimeData.segments;
+      }
+      if (!content && realTimeData.content) {
+        content = realTimeData.content;
+      }
+    }
+    let impObj = {
+      id: bidId,
+      tagid: uid.toString(),
+      ext: {
+        divid: adUnitCode
+      }
+    };
+
+    if (!mediaTypes || mediaTypes[BANNER]) {
+      const banner = createBannerRequest(bid, mediaTypes ? mediaTypes[BANNER] : {});
+      if (banner) {
+        impObj.banner = banner;
+      }
+    }
+    if (mediaTypes && mediaTypes[VIDEO]) {
+      const video = createVideoRequest(bid, mediaTypes[VIDEO]);
+      if (video) {
+        impObj.video = video;
+      }
+    }
+
+    if (impObj.banner || impObj.video) {
+      imp.push(impObj);
+    }
+  });
+
+  const source = {
+    tid: auctionId,
+    ext: {
+      wrapper: 'Prebid_js',
+      wrapper_version: '$prebid.version$'
+    }
+  };
+
+  if (schain) {
+    source.ext.schain = schain;
+  }
```

Comment:
Update your docs, add information that you support schain.  
Also, you could add some test case that will cover this branch.